src/extrapolation/extrap_prep.jl

```diff
@@ -0,0 +1,138 @@
+"""
```

Comment:
I now removed all the one-liners that explained each method separately (I figure that understanding can be reconstructed with `@which`, if one really wants it) and added a more general overview of how these methods are constructed to work together, and what one needs to implement for new schemes. Is this understandable for someone who didn't write the code?